checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -43,104 +82,596 @@ public class MustCallInferenceLogic {
+        } else if (node instanceof AssignmentNode) {
+          updateObligationsForAssignment(obligations, (AssignmentNode) node);
         }
       }
 
-      propagateRegPaths(current, visited, worklist);
+      propagateRegPaths(obligations, current.block, visited, worklist);
     }
   }
 
   /**
-   * If the receiver of {@code mNode} is a candidate owning field and the method invocation
-   * satisfies the field's must-call obligation, then adds that field to the {@link #owningFields}
-   * set.
+   * Returns a set of obligations representing the non-empty MustCall parameters of the current
+   * method.
    *
-   * @param mNode the MethodInvocationNode
+   * @param cfg the control flow graph of the method to check
+   * @return a set of obligations representing the non-empty MustCall parameters of the current
+   *     method
+   */
+  private Set<Obligation> getNonEmptyMCParams(ControlFlowGraph cfg) {
+    // TODO what about lambdas?
+    if (cfg.getUnderlyingAST().getKind() != UnderlyingAST.Kind.METHOD) {
+      return Collections.emptySet();
+    }
+    Set<Obligation> result = new LinkedHashSet<>(1);
+    for (VariableTree param : enMethodTree.getParameters()) {
+      VariableElement paramElement = TreeUtils.elementFromDeclaration(param);
+      if (typeFactory.declaredTypeHasMustCall(param)) {
+        result.add(
+            new Obligation(
+                ImmutableSet.of(new ResourceAlias(new LocalVariable(paramElement), param))));
+      }
+    }
+    return result;
+  }
+
+  /**
+   * This function returns a set of all owning fields that have been inferred in the current or
+   * previous iteration
+   *
+   * @return set of owning fields
+   */
+  private Set<VariableElement> getEnclosedOwningFields() {
+    ClassTree classTree = TreePathUtil.enclosingClass(typeFactory.getPath(enMethodTree));
+    TypeElement classElt = TreeUtils.elementFromDeclaration(classTree);
+    Set<VariableElement> enOwningFields = new HashSet<>();
+    for (Element memberElt : classElt.getEnclosedElements()) {
+      if (memberElt.getKind().isField() && typeFactory.hasOwning(memberElt)) {
+        enOwningFields.add((VariableElement) memberElt);
+      }
+    }
+    for (Element memberElt : owningFieldToECM) {
+      enOwningFields.add((VariableElement) memberElt);
+    }
+    return enOwningFields;
+  }
+
+  /**
+   * Given an index, adds an owning annotation to the parameter at the specified index
+   *
+   * @param index index of enclosing method's parameter
+   */
+  private void addOwningOnParams(int index) {
+    WholeProgramInference wpi = typeFactory.getWholeProgramInference();
+    wpi.addDeclarationAnnotationToFormalParameter(enMethodElt, index, OWNING);
+  }
+
+  /**
+   * Checks if the given node is a field and if it is a new Owning field.
+   *
+   * @param node the possible owning field
+   * @param mNode method invoked on the possible owning field
+   */
+  private void isOwningField(@Nullable Node node, MethodInvocationNode mNode) {
```

Comment:
The name and documentation of this method don't align with its type: methods named using the pattern `isNoun` should always return `boolean` (i.e., they should answer the question that their name poses). This method doesn't do that: instead, it seems like it infers owning for the given node, if the node is a field?

The documentation and name should both be rewritten to represent what this method actually accomplishes.

Reply:
You're correct! I have updated the name to inferOwningField and made corresponding changes to the documentation.